File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/enhetsregisteret/Underenhet.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.enhetsregisteret;

import lombok.Builder;
import lombok.Data;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.domene.Orgnr;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.domene.Orgnr;

@Builder
@Data
public class Underenhet implements Virksomhet {
    private final Orgnr orgnr;
    private final Orgnr overordnetEnhetOrgnr;
    private final String navn;
    private final Næringskode5Siffer næringskode;
    private final int antallAnsatte;
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domene/Underenhet.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domene

sealed class Underenhet {
    abstract val orgnr: Orgnr
    abstract val overordnetEnhetOrgnr: Orgnr
    abstract val navn: String
    abstract val antallAnsatte: Int

    data class Næringsdrivende(
        override val orgnr: Orgnr,
        override val overordnetEnhetOrgnr: Orgnr,
        override val navn: String,
        override val næringskode: Næringskode5Siffer,
        override val antallAnsatte: Int
    ) : Underenhet(), Virksomhet

    data class IkkeNæringsdrivende(
        override val orgnr: Orgnr,
        override val overordnetEnhetOrgnr: Orgnr,
        override val navn: String,
        override val antallAnsatte: Int
    ) : Underenhet()
}